### PR TITLE
Rating Feature

### PR DESCRIPTION
**Pull request by PederSchacht**
Ratings work  and update with page refreshing. Ajax to be implemented to allow only the rating refresh when a new rating is added or updated.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,6 +1,19 @@
 class Post < ActiveRecord::Base
   belongs_to :user
+  has_many :ratings
 
   validates_presence_of :content
   validates_presence_of :user
+
+  def rating_count(opinion)
+    if opinion == 'negative'
+      ratings.where(opinion: -1).count
```
Comment (elizamarcum) [inline app/models/post.rb]: I'd prefer that you wrap the business logic of what makes a rating positive/neutral/negative into scopes on the ratings model.

So, something like `scope :positive, ->{ where(opinion: -1) }

If you do that, you could actually refactor this to be

ratings.send(opinion).count

Which would be super cool!

File: app/views/posts/_ratings.haml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+- @rating = Rating.where(:user_id => current_user, :post_id => post).first
```
Comment (elizamarcum) [inline app/views/posts/_ratings.haml]: I think that this should be:

`- rating = current_user.ratings.where(post: post).first || current_user.ratings.build`

That will allow you to combine the two cases below into a single set of forms.

**Comment by elizamarcum** [comment]
There are a few refactors to do, but this looks good overall.

**Comment by PederSchacht** [comment]
Ratings update with ajax and nonessential (pending) tests for rating update removed.

**Comment by PederSchacht** [comment]
requested refactors completed.

**Pull request closed without merging**